src/components/RelicModal.js

```diff
@@ -249,7 +271,35 @@ export default function RelicModal(props) {
     props.setOpen(false)
   };
   const onValuesChange = (x) => {
-    console.log('Form change', x);
+    let mainStatOptions = [];
+    if (x.part) {
+      mainStatOptions = Object.entries(Constants.PartsMainStats[x.part]).map(entry => ({
+        label: entry[1],
+        value: entry[1]
+      }));
+      setMainStatOptions(mainStatOptions);
+      relicForm.setFieldValue('mainStatType', mainStatOptions[0]?.value);
+    }
+
+    let mainStatType = mainStatOptions[0]?.value || relicForm.getFieldValue('mainStatType');
+    let enhance = relicForm.getFieldValue('enhance');
+    let grade = relicForm.getFieldValue('grade');
+
+    if (mainStatType != undefined && enhance != undefined && grade != undefined) {
+      const specialStats = [Constants.Stats.OHB, Constants.Stats.Physical_DMG, Constants.Stats.Physical_DMG, Constants.Stats.Fire_DMG, Constants.Stats.Ice_DMG, Constants.Stats.Lightning_DMG, Constants.Stats.Wind_DMG, Constants.Stats.Quantum_DMG, Constants.Stats.Imaginary_DMG];
+      const floorStats = [Constants.Stats.HP, Constants.Stats.ATK, Constants.Stats.SPD];
+
+      let mainStatValue = Constants.MainStatsValues[mainStatType][grade]['base'] + Constants.MainStatsValues[mainStatType][grade]['increment'] * enhance;
+
+      if (specialStats.includes(mainStatType)) { // Outgoing Healing Boost and elemental damage bonuses has a weird rounding with one decimal place
+        mainStatValue = Math.floor(mainStatValue * 10) / 10;
```

Comment:
@ronaldpereira Just for future reference I have a couple util functions that help for these cases

```
  precisionRound(number, precision = 8) {
    let factor = Math.pow(10, precision);
    return Math.round(number * factor) / factor;
  },
...
  truncate10ths: (x) => {
    return Math.floor(x * 10) / 10
  },
 ```

Reply:
didn't see that when I was implementing it
we'll update this relic modal soon to implement more UI changes like we were discussing in discord, so I'll keep this in mind to use this functions, thanks!